### PR TITLE
fix(platform): multi-input issues

### PR DESCRIPTION
**Pull request by dmitry-stepanenko**
## Related Issue(s)

part of defect hunting sprint 77 https://github.com/SAP/fundamental-ngx/issues/7362
issue has been raised in comment https://github.com/SAP/fundamental-ngx/issues/7362#issuecomment-993928134

## Description
- Platform multi-input mobile mode - button on right of input resets the search list but does not reset input. Also, "approve" button in dialog footer needs an uppercase "A". Lastly the top right X button does not close the dialog
- Platform multi-input decline list type - list options lose their X on subsequent opens of the popover
- when "close" is clicked, input values will be reset
-  null invalidity should style the input red on blur.
- added CVA unit tests

##### During Implementation

1. Visual Testing:

-   [x] visual misalignments/updates
-   [n/a] check Light/Dark/HCB/HCW themes
-   [n/a] RTL/LTR - proper rendering and labeling
-   [n/a] responsiveness(resize)
-   [n/a] Content Density (Cozy/Compact/(Condensed))
-   [x] States - hover/disabled/focused/active/on click/selected/selected hover/press state
-   [x] Interaction/Animation - open/close, expand/collapse, add/remove, check/uncheck
-   [x] Mouse vs. Keyboard support
-   [n/a] Text Truncation

2. API and functional correctness

-   [x] check for console logs (warnings, errors)
-   [x] API boundary values
-   [n/a] different combinations of components - free style
-   [x] change the API values during testing

3. Documentation and Example validations

-   [n/a] missing API documentation or it is not understandable
-   [n/a] poor examples
-   [n/a] Stackblitz works for all examples

4. Accessibility testing
5. Browser Testing - Edge, Safari, Chrome, Firefox

##### PR Quality

-   [x] the commit message(s) follows the guideline:
        https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
-   [x] tests for the changes that have been done
-   [n/a] all items on the PR Review Checklist are addressed :
        https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
-   [n/a] Run npm run build-pack-library and test in external application
-   [n/a] update `README.md`
-   [n/a] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)

### CODE REVIEW

**Review by N1XUS — APPROVED** [review]

**Comment by dimamarksman** [comment]
there is nodeployed preview

**Comment by dmitry-stepanenko** [comment]
@N1XUS pushed more changes (separate commit), please take a look again

File: libs/platform/src/lib/form/multi-input/multi-input.component.html [hunk 1]
```diff
@@ -37,8 +37,7 @@
         [disabled]="disabled"
         [compact]="contentDensity === 'compact'"
         (keydown)="removeSelectedTokens($event)"
-        (addOnButtonClicked)="!mobile && addOnButtonClick()"
-        (click)="mobile && onPrimaryButtonClick(false)"
```
Comment (dimamarksman) [inline libs/platform/src/lib/form/multi-input/multi-input.component.html:41]: @dmitry-stepanenko in mobile mode clicking on input should open full-screen window, currently I don't see this

Comment (dmitry-stepanenko) [reply libs/platform/src/lib/form/multi-input/multi-input.component.html:41]: @dimamarksman  fixed

**Review by Betrozov — APPROVED** [review]

**Review by dimamarksman — APPROVED** [review]

**Review by N1XUS — APPROVED** [review]

**Review by mikerodonnell89 — COMMENTED** [review]
Just wondering, now I'm seeing that the icon button in the input seems to dismiss the dialog the same way as the accept button, is this the appropriate icon? Should there even be an icon? Unclear to me as a user what this icon is supposed to do

**Comment by dmitry-stepanenko** [comment]
> Just wondering, now I'm seeing that the icon button in the input seems to dismiss the dialog the same way as the accept button, is this the appropriate icon? Should there even be an icon? Unclear to me as a user what this icon is supposed to do

hi @mikerodonnell89, the button does the same thing as with the regular multi-input - it toggles the list of options. I wasn't able to find any specs on it's behavior in mobile mode, so I decided to stick to how it works in `core-multi-input`

**Review by mikerodonnell89 — APPROVED** [review]

**Pull request merged**